libs/langgraph/langgraph/managed/context.py

```diff
@@ -0,0 +1,85 @@
+from contextlib import asynccontextmanager, contextmanager
+from inspect import signature
+from typing import (
+    Any,
+    AsyncContextManager,
+    AsyncIterator,
+    ContextManager,
+    Iterator,
+    Optional,
+    Self,
+    Type,
+    Union,
+)
+
+from langchain_core.runnables import RunnableConfig
+
+from langgraph.managed.base import ConfiguredManagedValue, ManagedValue, V
+
+
+class Context(ManagedValue):
+    runtime = True
+
+    value: V
+
+    @staticmethod
+    def of(
+        ctx: Union[None, Type[ContextManager[V]], Type[AsyncContextManager[V]]] = None,
+        actx: Optional[Type[AsyncContextManager[V]]] = None,
+    ) -> ConfiguredManagedValue:
+        if ctx is None and actx is None:
+            raise ValueError("Must provide either sync or async context manager.")
+        return ConfiguredManagedValue(Context, {"ctx": ctx, "actx": actx})
+
+    @classmethod
+    @contextmanager
+    def enter(cls, config: RunnableConfig, **kwargs: Any) -> Iterator[Self]:
+        with super().enter(config, **kwargs) as self:
+            if self.ctx is None:
+                raise ValueError("Cannot enter sync context manager.")
```

Comment:
Out of context, this could be confusing. Perhaps: 

```suggestion
                raise ValueError("Synchronous context manager not found. Please initialize Context value with a sync context manager, or invoke your graph asynchronously.")
```

or something less verbose but more contextual